File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> ans;
    void recur(vector<int> nums, vector<int> temp, int idx){
        if (idx == nums.size()){
            ans.push_back(temp);
            return;
        }
        recur(nums, temp, idx+1);
        temp.push_back(nums[idx]);
        recur(nums, temp, idx+1);
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<int> temp;
        int idx =0;
        ans.clear();
        recur(nums,temp,idx);
        return ans;
        
    }
};

File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []

        def recur(temp, idx):
            if idx== len(nums):
                res.append(temp)
                return 
            recur(temp, idx+1)
            recur(temp+[nums[idx]], idx+1)
        recur([],0)
        return res
            